### PR TITLE
Node 10 support

### LINKED ISSUE #1813: Support Node 10. WAS: Realm fails to install via yarn or npm, with the error Failed to execute 'node-gyp build --fallback-to-build --module=[hidden_path]/node_modules/realm/compiled/node-v64_darwin_x64/realm.node --
## Goals

Use yarn or npm to install realm

## Expected Results

Installs correctly

## Actual Results

404 on https://static.realm.io/node-pre-gyp/2.3.2/realm-v2.3.2-node-v64-darwin-x64.tar.gz which causes install to fail

## Version of Realm and Tooling

- Realm JS SDK Version: 2.3.2
- Node or React Native: React Native
- Client OS & Version: Mac 10.13.4
- Which debugger for React Native: None

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

Adding support for node 10, and removing node 7 and 9 support (odd-numbered versions of node are development versions).

This closes #1813

## ☑️ ToDos

* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* ~[ ] 🚦 Tests~
* ~[ ] 📝 Public documentation PR created or is not necessary~
* ~[ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
* ~[ ] typescript definitions file is updated~
* ~[ ] jsdoc files updated~
* ~[ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: realm.gypi [hunk 1]
```diff
@@ -198,9 +198,14 @@
     {
       "target_name": "realm-sync",
       "type": "none",
-      "dependencies": [ "realm-core" ],
+      "dependencies": [ "realm-core", "OpenSSL" ],
       "link_settings": {
           "libraries": [ "-lrealm-sync<(debug_library_suffix)" ],
+          "conditions": [
+              ["OS=='win'", {
+                "libraries": [ "version.lib" ]
```
Comment (fealebenpae) [inline realm.gypi]: Why do we need to link against this?

File: realm.gypi [hunk 2]
```diff
@@ -239,7 +244,8 @@
           }],
           ["OS=='linux'", {
             "libraries": [ "-l:libssl.a", "-l:libcrypto.a" ],
-            "library_dirs": [ "/usr/lib", "/usr/lib64" ],
+              "library_dirs": [ "/usr/lib", "/usr/lib64" ],
```
Comment (fealebenpae) [inline realm.gypi]: Indentation seems to be off.

File: realm.gypi (hunk 1 above)
Comment (kneth) [reply realm.gypi]: There is a call to `GetFileVersionInfoExW` in Realm Sync: https://github.com/realm/realm-sync/blob/c62edffc9411119d3cb45768d33f939a2a44ea5e/src/realm/util/platform_info.cpp#L102

File: realm.gypi [hunk 3]
```diff
@@ -201,6 +202,17 @@
       "dependencies": [ "realm-core" ],
       "link_settings": {
           "libraries": [ "-lrealm-sync<(debug_library_suffix)" ],
+          "conditions": [
+              ["OS=='win'", {
+                "conditions": [
+                  ["target_arch=='ia32'", {
+                    "libraries": [ "C:\\Program Files (x86)\\Windows Kits\\8.1\\Lib\\winv6.3\\um\\x86\\mincore.lib" ]
+                  }, {
+                    "libraries": [ "C:\\Program Files (x64)\\Windows Kits\\8.1\\Lib\\winv6.3\\um\\x64\\mincore.lib" ]
```
Comment (fealebenpae) [inline realm.gypi]:
```suggestion
                    "libraries": [ "C:\\Program Files (x86)\\Windows Kits\\8.1\\Lib\\winv6.3\\um\\x64\\mincore.lib" ]
```

It's always `C:\Program Files (x86)`

Comment (kneth) [reply realm.gypi]: Thanks.

File: packaging/node-pre-gyp/base-image/Dockerfile [hunk 4]
```diff
@@ -18,7 +18,7 @@ RUN touch /var/lib/rpm/* \
     && yum -y install \
         python27 \
         which \
-        chrpath \
+        chrpath \:q
```
Comment (fealebenpae) [inline packaging/node-pre-gyp/base-image/Dockerfile]: This looks like a vim artefact

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,7 @@ x.x.x Release notes (yyyy-MM-dd)
 =============================================================
 ### Enhancements
 * A more meaningful exception will be thrown when trying to refresh the access token for a Realm with an invalid url. Previously, trying to connect to a Realm with a url that lacks the path component (e.g. `realm://foo.com`) would result in errors like `Cannot read property ‘token_data’ of undefined`. Instead, now we'll print out the Realm url and provide a more meaningful exception message. ([#ROS-1310](https://github.com/realm/realm-object-server-private/issues/1310), since v1.0.2)
+* Adds support for Node 10. Pre-gyp'ed binaries are avaiable for Node 6, 8, and 10. ([#1813](https://github.com/realm/realm-js/issues/1813) and [#2087](https://github.com/realm/realm-js/issues/2087))
```
Comment (cmelchior) [inline CHANGELOG.md]: avaiable -> available

**Pull request merged**